index.js

```
@@ -0,0 +1,132 @@
+        } 
+        
+        if(req.msg && !req.msg.startsWith('/')) {
+            cb(null, true) /* Yes I am handling this message */
+            reqChatGPTServer(req.msg,(err,cgptAnswer)=>{
+                if(err){
+                    sendReply("Oops, it looks like something went wrong. Could you please try again.", req)
+                    u.showErr(err.message)
+                }
+                else sendReply(cgptAnswer,req);
+            })
+       
+        } else cb() /* REMEMBER TO CALL THIS OTHERWISE THE AVATAR WILL WAIT FOR A RESPONSE FOREVER */ 
+    })
+
+    async function reqChatGPTServer(msg,cb) {
+        
+        const instance = axios.create({
+            httpsAgent: new https.Agent({  
+              rejectUnauthorized: false
+            })
+          });
+        
+        var body = {
+            model: sModel,
+            prompt: msg,
+            max_tokens: iMaxTokens,
+            user: sUserId,
+            temperature:  dTemperature,
+            frequency_penalty: 0.0, //Number between -2.0 and 2.0  Positive value decrease the model's likelihood to repeat the same line verbatim.
+            presence_penalty: 0.0,  //Number between -2.0 and 2.0. Positive values increase the model's likelihood to talk about new topics.
+            stop: ["#", ";"] //Up to 4 sequences where the API will stop generating further tokens. The returned text will not contain the stop sequence.
+        }
+        instance.post('https://api.openai.com/v1/completions', body, {
+            headers : headers
+        })
+        .then(function(response){
+            if(response.data.choices && response.data.choices.length > 0){
+                const result = response.data.choices;
+                const chatGPTAnswer = result[0].text
+                cb(null, chatGPTAnswer.trim());
+            }
+            else if(response.data.error){
+                u.showErr(response.data.error)
+                cb(response.data.error)
+            }else cb()
```

This is also an "error" from our perspective - we did not get the response we expected and so we should return an error accordingly (`cb("Unexpected response: Did not find 'choices' or 'error' in response")`)